charts/tfy-agent/values.yaml

```diff
@@ -1,123 +1,296 @@
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: "a9af8a51d8b0373f4d0c469119ed88a488c5709f"
+
+  ## Define resources requests and limits for single Pods.
+  ## ref: https://kubernetes.io/docs/user-guide/compute-resources/
+  ##
+  ## customized resource limits are very general in nature and must be kept the same
+  ## decreasing the resource might make the service UNAVAILABLE
+  resources:
+    limits:
+      cpu: 500m
+      memory: 512Mi
+      ephemeral-storage: 256Mi
+    requests:
+      cpu: 300m
+      memory: 256Mi
+      ephemeral-storage: 128Mi
 
-podSecurityContext: {}
-  # fsGroup: 2000
+  livenessProbe:
+    failureThreshold: 5
+    httpGet:
+      path: /
+      port: 3000
+      scheme: HTTP
+    periodSeconds: 10
+    successThreshold: 1
+    timeoutSeconds: 1
+    initialDelaySeconds: 15
 
-securityContext: {}
+  readinessProbe:
+    failureThreshold: 5
+    httpGet:
+      path: /
+      port: 3000
+      scheme: HTTP
+    periodSeconds: 10
+    successThreshold: 1
+    timeoutSeconds: 1
+    initialDelaySeconds: 15
+  
+  ## SecurityContext holds pod-level security attributes and common container settings.
+  ## This defaults to non root user with uid 1000 and gid 2000. *v1.PodSecurityContext  false
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
+  ##
+  securityContext: {}
   # capabilities:
   #   drop:
   #   - ALL
   # readOnlyRootFilesystem: true
   # runAsNonRoot: true
   # runAsUser: 1000
 
-service:
-  type: ClusterIP
-  port: 3000
-  annotations: {}
-
+  ## If specified, the pod's tolerations.
+  ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  ##
+  tolerations: []
+  # - key: "key"
+  #   operator: "Equal"
+  #   value: "value"
+  #   effect: "NoSchedule"
+    
+  ## Define which Nodes the Pods are scheduled on.
+  ## ref: https://kubernetes.io/docs/user-guide/node-selection/
+  ##
+  nodeSelector: {}
 
-ingress:
-  enabled: false
-  annotations: {}
-  labels: {}
-  ingressClassName: istio
-  tls: []
-  hosts: []
-
-resources:
-  limits:
-    cpu: 200m
-    memory: 512Mi
-    ephemeral-storage: 256Mi
-  requests:
-    cpu: 100m
-    memory: 300Mi
-    ephemeral-storage: 128Mi
+  ## Assign custom affinity rules to the tfyAgent instance
+  ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+  ##
+  affinity: {}
+  # nodeAffinity:
+  #   requiredDuringSchedulingIgnoredDuringExecution:
+  #     nodeSelectorTerms:
+  #     - matchExpressions:
+  #       - key: kubernetes.io/e2e-az-name
+  #         operator: In
+  #         values:
+  #         - e2e-az1
+  #         - e2e-az2
 
+# configuration for tfy-agent-proxy
+tfyAgentProxy:
 
-healthcheck:
+  ## Deploy tfyAgentProxy
+  #
   enabled: true
-  readiness:
-    port: 3000
-    path: /
-  liveness:
-    port: 3000
-    path: /
 
-nodeSelector: {}
-tolerations: {}
-affinity: {}
+  ## Number of replicas of tfyAgent
+  #
+  replicaCount: 1
```

Comment:
We should not expose this option for `tfy-agent`/`tfy-agent-proxy` as of now. There should be strictly only 1 pod for these services.

Reply:
Later if we want to implement a hot stand-by pod via leader election, we can expose this.